File: ted-ext/ted-scheduler/src/main/java/labai/ted/sys/TedSchdDriverExt.java
package labai.ted.sys;


import labai.ted.TedDriver;
import labai.ted.TedSchdHck1;

import javax.sql.DataSource;

/**
 * @author Augustus
 *         created on 2018.08.17
 *
 *  use some of internal functions.
 *
 *  For ted-scheduler only!
 *
 */
public class TedSchdDriverExt {
	private final TedDriver tedDriver;
	private final TedDriverImpl tedDriverImpl;

	public TedSchdDriverExt(TedDriver tedDriver) {
		this.tedDriver = tedDriver;
		this.tedDriverImpl = TedSchdHck1.getTedDriverImpl(tedDriver);
	}

	public String systemId() {
		return tedDriverImpl.getContext().config.systemId();
	}
	public String instanceId() {
		return tedDriverImpl.getContext().config.instanceId();
	}
	public Long primeTaskId() {
		return tedDriverImpl.getContext().prime.primeTaskId();
	}
	public DataSource dataSource() {
		return tedDriverImpl.dataSource;
	}
	public boolean isPrimeEnabled() {
		return tedDriverImpl.prime().isEnabled();
	}
}

File: ted-ext/ted-scheduler/src/main/java/ted/scheduler/impl/TedSchdDriverExt.kt
package ted.scheduler.impl


import ted.driver.TedDriver
import ted.driver.sys.SqlUtils.DbType
import ted.driver.sys._SchdExt
import javax.sql.DataSource

/**
 * @author Augustus
 * created on 2018.08.17
 *
 * use some of internal functions.
 *
 * For ted-scheduler only!
 */
class TedSchdDriverExt(tedDriver: TedDriver) : _SchdExt(tedDriver) {
    public override fun isPrimeEnabled() = super.isPrimeEnabled()
    public override fun systemId(): String = super.systemId()
    public override fun instanceId(): String = super.instanceId()
    public override fun primeTaskId(): Long? = super.primeTaskId()
    public override fun dataSource(): DataSource = super.dataSource()
    public override fun dbType(): DbType = super.dbType()
    public override fun tableName(): String = super.tableName()
    public override fun schemaName(): String? = super.schemaName()
    public override fun getPropertyValue(key: String): String? = super.getPropertyValue(key)
}
